src/core/trulens/core/tru.py

```diff
@@ -821,6 +829,81 @@ def get_leaderboard(
                 .sort_values(by=feedback_cols, ascending=False)
             )
 
+    def add_ground_truth_nowait(
+        self,
+        ground_truth: mod_groundtruth_schema.GroundTruth,
+    ) -> None:
+        """Add a ground truth object to the queue to be inserted in the next batch."""
+        if self.batch_thread is None:
+            self.batch_thread = threading.Thread(
+                target=self.batch_ground_truth_loop, daemon=True
+            )
+            self.batch_thread.start()
+        self.batch_ground_truth_queue.put(ground_truth)
+
+    def batch_ground_truth_loop(self):
```

Comment:
We probably don't need to do this in a separate thread, instead we can do an in-memory buffer and write in the same thread as `add_ground_truth_to_dataset` but do a batch insert.

Reply:
updated based on advice